### PR TITLE
BaSP-TG-4: Super Admins.

### PR DESCRIPTION
**Pull request by lucianofrailelopez**
This PR contains the following methods:
1.GetAllSuperAdmin.
2. Get SuperAdmin by id.
3.Create an SuperAdmin.
4.Deleting an SuperAdmin.
5.updating an SuperAdmin.
6. Validation using Joi.

### CODE REVIEW

File: src/controllers/super-admins.js [hunk 1]
```diff
@@ -1,121 +1,104 @@
...
+      return res.status(404).json({
+        msg: 'The admin has not been found',
+      });
+    }
+    return res.status(200).json({
+      msg: 'The admin has been successfully deleted',
+    });
+  } catch (error) {
+    return res.json({
+      msg: 'An error has ocurred',
```
Comment (EzeRocco) [inline src/controllers/super-admins.js:48]: error parameter is missing

File: src/controllers/super-admins.js [hunk 2]
```diff
@@ -1,121 +1,104 @@
...
-          created_superadmin: newSuperAdmin,
-        });
-      }
+const getAllSuperAdmins = async (req, res) => {
+  try {
+    const AllSuperAdmins = await superAdminModel.find({});
+    return res.status(200).json(AllSuperAdmins);
+  } catch (error) {
+    return res.status(500).json({
+      msg: 'There was an error',
```
Comment (ArianaMaldonado) [inline src/controllers/super-admins.js:9]: Return should contain msg, data and error.

File: src/controllers/super-admins.js [hunk 3]
```diff
@@ -1,121 +1,104 @@
...
+    if (req.params.id) {
+      const superAdmin = await superAdminModel.findById({ _id: req.params.id });
+      return res.status(200).json(superAdmin);
+    }
+    return res.status(400).json({
+      msg: 'missing id parameter',
+    });
+  } catch (error) {
+    return res.json({
+      msg: error,
```
Comment (ArianaMaldonado) [inline src/controllers/super-admins.js:25]: same here

Comment (ArianaMaldonado) [reply src/controllers/super-admins.js:25]: Same in all returns

**Review by EzeRocco — APPROVED** [review]

**Review by ArianaMaldonado — APPROVED** [review]

**Pull request merged**